keps/sig-cli/859-kubectl-headers/README.md

```diff
@@ -257,11 +265,161 @@ included.
+can be no version skew.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+
+  This feature is enabled with the KUBECTL_COMMANDS_HEADER environment
+  variable set on the client command line. Since it only affects the client
+  kubectl, it does not affect any cluster components.
+
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: kubectl-commands-in-headers
+    - Components depending on the feature gate: kubectl
+  - [X] Other
+    - Describe the mechanism: setting an client-side environment variable for kubectl.
+    - Will enabling / disabling the feature require downtime of the control
+      plane? No
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      No.
+
+* **Does enabling the feature change any default behavior?**
+
+  No. This feature is not user facing, so it does not change behavior.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  Yes. This feature can be disabled by simply removing an environment variable
+  on the client command line.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  Re-enabling this feature is simply accomplished by setting the feature
+  environment variable on the client command line. There is no state, and there
+  is no consequence for re-enabling the feature.
+
+* **Are there any tests for feature enablement/disablement?**
+
+  There will be unit tests and integration tests which test this
+  feature enablement and disablement by setting the environment
+  variable.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+  This feature only affects kubectl; not any cluster components. So it would not
+  be possible to impact running workloads.
+
+* **What specific metrics should inform a rollback?**
+
+  Not applicable.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+
+  Enabling and disabling the client-side command line environment variable will
+  be tested.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+
+  No
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+  checking if there are objects with field X set) may be a last resort. Avoid
+  logs or events for this purpose.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [ ] Metrics
```

Comment:
Seeing whether the feature is being used is pretty valuable for deciding its future. In this case, it seems reasonable to wire a metric to at least the commands being used.

The cardinality on a lot of the data you're gathering seems excessive for eventual use, but the command names don't look too bad.

Reply:
We definitely intend to have our opt-in alpha users provide feedback on this mechanism. So we will fill out the `Metrics` section upon our move to beta as the template suggests.

Reply:
In the majority of cases the feature is driven by the flags beause the "is this a subcommand vs a flag on an existing command" is relatively arbitrary.  Flags such as `--prune` for kubectl apply or `-f / -k` for delete are critical to understanding the usage.